docs/external-repositories-guides/maintainer-guide.md

```diff
@@ -0,0 +1,74 @@
+# Guide for Repository maintainers
+
+This page explains how to install and configure your own custom Deadlock challenges repository.
+
+## Create your *GitLab* repository
+
+First of all, you will need a git repository to store your challenges. Deadlock currently only supports *GitLab* hosted repositories (it can be on a private instance or on the public *gitlab.com* hosting).
+
+The repository doesn't have to be public, you can very well make it private if you wish so.
+
+We recommend [this template](https://git.e-biz.fr/deadlock-public/deadlock-challenges-example) to start with as it comes directly with some useful scripts and a CI working out of the box (see below).
+
+> **Important**: Deadlock pulls challenges from the **master** branch, so please make sure you have one.
+
+## Add your challenges to your *GitLab* repository
+
+Now you can populate the **resources** folder with your challenges. If you are not familiar with the process of creating and testing your challenges with **DCLI**, you can visit our [challenge documentation](../index.md) page.
+
+
+## Build and push your challenge images to your container registry
+
+
+If you had our template repository as a starting point, then you have a working out of the box `.gitlab-ci.yml` which will automatically build and push your challenges to your registry.
+
+If you prefer doing it manually, you can do it with the following commands:
+
+* login into your registry
+
+```bash
+docker login registry.gitlab.com -u <username> -p <token>
+```
+
+
+* build  and push a challenge image
+
+```bash
+cd resources/my_challenge
+docker build -t registry.gitlab.com/your_group/project_name/my_challenge:version .
+docker push registry.gitlab.com/your_group/project_name/my_challenge:version
+```
+
+> **Important**: *my_challenge:version* has to correspond to mission name and version from *challenge.yml*.
+
+You can visit this [gitlab documentation](https://docs.gitlab.com/ee/user/packages/container_registry/) page for more details.
+
+## Make your repository available for a Deadlock instance
+
+In order to your challenges become visible and runnable from a Deadlock instance, you need to add their SSH key as a deploy key and provide the Deadlock platform an authentication *username:password* pair.
```

Comment:
In order to make your challenge visible ...

provide the Deadlock plateform **with**